src/thor/bidirectional_astar.cc

```diff
@@ -357,8 +357,13 @@ inline bool BidirectionalAStar::ExpandInner(baldr::GraphReader& graphreader,
 
   // setting this edge as reached
   if (expansion_callback_) {
-    expansion_callback_(graphreader, FORWARD ? meta.edge_id : opp_edge_id, "bidirectional_astar", "r",
-                        pred.cost().secs, pred.path_distance(), pred.cost().cost);
+    const auto& prev_pred =
+        pred.predecessor() == kInvalidLabel
+            ? GraphId{}
+            : (FORWARD ? edgelabels_forward_ : edgelabels_reverse_)[pred.predecessor()].edgeid();
+    expansion_callback_(graphreader, FORWARD ? meta.edge_id : opp_edge_id, prev_pred,
+                        "bidirectional_astar", "r", newcost.secs,
+                        pred.path_distance() + meta.edge->length(), newcost.cost);
```

Comment:
I also changed all callbacks to use the end of the current edge. Most were doing that already, but not the "reached" ones. That's really confusing when looking at the results.